### PR TITLE
Necessary Updates

### PR DESCRIPTION
**Pull request by prertik**
WIP. Still is a long way off. :sweat_smile: 

- [ ] #2265
- [ ] #1973
  - [x] Remove uses of `lt.util.cljs/clj->js` and `lt.util.cljs/js->clj` and rely on the fns of the same name that come with clojurescript
  - [x] Fix all compilation warnings that show up on latest `lein cljsbuild`. Known warnings to fix include missing requires and prefix JSON e.g. js/JSON
  - [x] Ability to recompile cljsDeps.js after a cljs upgrade with one command [like this](https://github.com/BenjaminVanRyseghem/LightTable/commit/e2fffcc9d7e58ddaf48fa58cfaeb3f263136956f)
  - [x] Upgrade to latest cljs and lein-cljsbuild. (Branch currently using cljs 1.7.145, clj 1.7.0, and lein-cljsbuild 1.1.4)
  - [x] Recompile cljsDeps.js for upgraded cljs: `lein cljsbuild once cljsdeps`
  - [x] Make sure extend-types that occur on core classes are updated - [see cljs.cljs](https://github.com/LightTable/LightTable/blob/master/src/lt/util/cljs.cljs)
  - [ ] Fix Clojure(Script) eval. Does not work for to be investigated reasons
  - [x] Fix threading issue when starting LT: `arguments$ is not defined`.
  - [ ] Fix erratic printing with `prn`. Trying to prn a LT object sometimes doesn't show up
  - [x] Tag and release fetch 3.0 for https://github.com/LightTable/fetch/pull/17
  - [ ] Get plugins working again (e.g., Rainbow is not rainbowing and workspace-nav is not naving)
  - [ ] Further upgrade cljs and lein-cljsbuild to latest available.
- [ ] #2365
- [ ] #2259

### CODE REVIEW

File: deploy/core/main.js [hunk 1]
```diff
@@ -131,26 +133,26 @@ function start() {
...
+    let oldSend = window.webContents.send;
     window.webContents.send = function () {
       console.log("\t\t\t\t\tMAIN->", Array.prototype.slice.call(arguments).join(', '));
       oldSend.apply(window.webContents, arguments);
     };
   };
-  var oldCreateWindow = createWindow;
-  var createWindow = function() { logSend(oldCreateWindow()); };
+  let oldCreateWindow = createWindow;
+  let createWindow = function() { logSend(oldCreateWindow()); };
```
Comment (Mouvedia) [inline deploy/core/main.js]: I am not sure you want `let` there.

File: deploy/core/package.json [hunk 2]
```diff
@@ -1,29 +1,30 @@
 {
   "name": "LightTable",
   "description": "LightTable Electron App",
-  "version": "0.8.1",
+  "version": "0.9.0",
```
Comment (Mouvedia) [inline deploy/core/package.json]: Ill delete the 0.8.2 milestone.

Comment (prertik) [reply deploy/core/package.json]: No, don't right now.

File: project.clj [hunk 3]
```diff
@@ -21,14 +22,14 @@
                         :compiler {:optimizations :simple
                                    :output-to "deploy/core/node_modules/clojurescript/cljsDeps.js"
                                    :output-dir "deploy/core/node_modules/clojurescript/cljsDeps/"
-                                   :pretty-print true }}]}
+                                   :pretty-print true}}]}
 
   ;; TODO: Remove separate :doc :dependencies after ClojureScript upgrade
```
Comment (Mouvedia) [inline project.clj:27]: What about that TODO?

File: README.md [hunk 4]
```diff
@@ -1,5 +1,4 @@
-# Light Table [![Build Status](https://travis-ci.org/LightTable/LightTable.svg?branch=master)](https://travis-ci.org/LightTable/LightTable)
-
+# Light Table [![Build Status](https://travis-ci.org/LightTable/LightTable.svg?branch=master)](https://travis-ci.org/LightTable/LightTable) [![CircleCI](https://circleci.com/gh/Dublo/LightTable-forked.svg?style=svg)](https://circleci.com/gh/Dublo/LightTable-forked)
```
Comment (Mouvedia) [inline README.md]: > Dublo/LightTable-forked

This is wrong. You will have to create an official account. In the meantime it will do.

Comment (prertik) [reply README.md]: Right. Forgot about it. I'll create a new PR on CI builds and other integrations for now...

File: deploy/core/node_modules/lighttable/browserInjection.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 (function(window) {
 
-  var ipc = require("ipc");
+  const { ipcRenderer } = require("electron");
```
Comment (Mouvedia) [inline deploy/core/node_modules/lighttable/browserInjection.js:4]: http://help.houndci.com/configuration/jshint
https://jshint.com/docs/options/

for reference here's my .jshintrc:
```
{
  "browser": true,
  "node": true,
  "moz": true,
  "eqeqeq": false,
  "expr": true,
  "lastsemic": true,
  "laxcomma": true,
  "eqnull": true,
  "proto": true,
  "nonew": true,
  "-W008": false,
  "-W030": false,
  "-W053": false,
  "-W056": false,
  "-W058": false,
  "-W086": true,
  "-W061": true,
  "-W067": true
}

**Comment by prertik** [comment]
This pull request **introduces 3 alerts** when merging f4d43654add35e9646d6d4f9bcbc22ae8718b065 into e2488885d7846a6d0ba203eb3975a3d39caff243 - [view on LGTM.com](https://lgtm.com/projects/g/LightTable/LightTable/rev/pr-0f512bed27d2e0f651b43ceeff983babf636960a)

**new alerts:**

* 1 for Useless assignment to local variable
* 1 for Invocation of non\-function
* 1 for Access to let\-bound variable in temporal dead zone

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by prertik** [comment]
This pull request **introduces 3 alerts** when merging ce4b6c5c7346e9310887910cfa5ced6a7c896b84 into e2488885d7846a6d0ba203eb3975a3d39caff243 - [view on LGTM.com](https://lgtm.com/projects/g/LightTable/LightTable/rev/pr-839fb935f1d4450d6f2067d40a89c34630e34753)

**new alerts:**

* 1 for Useless assignment to local variable
* 1 for Invocation of non\-function
* 1 for Access to let\-bound variable in temporal dead zone

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by prertik** [comment]
This pull request **introduces 3 alerts** when merging 81ffb8b9aa905698f1ad92447c26f37c952e6dee into e2488885d7846a6d0ba203eb3975a3d39caff243 - [view on LGTM.com](https://lgtm.com/projects/g/LightTable/LightTable/rev/pr-64655aec39fb22fb1a965ac332788bc836011c59)

**new alerts:**

* 1 for Useless assignment to local variable
* 1 for Invocation of non\-function
* 1 for Access to let\-bound variable in temporal dead zone

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: deploy/core/main.js [hunk 6]
```diff
@@ -152,7 +152,7 @@ if (process.env["IPC_DEBUG"]) {
       oldSend.apply(window.webContents, arguments);
     };
   };
-  let oldCreateWindow = createWindow;
+  let oldCreateWindow = createWindow();
```
Comment (Mouvedia) [inline deploy/core/main.js]: The `BrowserWindow` constructor returns an instance not a function so you don't want to add the parentheses.

Comment (prertik) [reply deploy/core/main.js]: Yes, this is a mistake. This was unintentional.

**Comment by prertik** [comment]
This pull request **introduces 3 alerts** when merging 2d0dba9946e6593340129965c3a8d43390af1770 into e2488885d7846a6d0ba203eb3975a3d39caff243 - [view on LGTM.com](https://lgtm.com/projects/g/LightTable/LightTable/rev/pr-9d15d95f1add38761e12161d1d5cf017203c5dbe)

**new alerts:**

* 1 for Useless assignment to local variable
* 1 for Invocation of non\-function
* 1 for Access to let\-bound variable in temporal dead zone

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by prertik** [comment]
This pull request **fixes 1 alert** when merging 1e4497b1f71579350fffa825d21865639ff1becd into e2488885d7846a6d0ba203eb3975a3d39caff243 - [view on LGTM.com](https://lgtm.com/projects/g/LightTable/LightTable/rev/pr-7e3ff1788de6708c2d5ab7728e51b0cd1f9a1e4b)

**fixed alerts:**

* 1 for Variable not declared before use

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by prertik** [comment]
This pull request **fixes 1 alert** when merging c5975d3d79f5c754f95f37024c552ebfba836bae into e2488885d7846a6d0ba203eb3975a3d39caff243 - [view on LGTM.com](https://lgtm.com/projects/g/LightTable/LightTable/rev/pr-a2b4c7cb751e3237279c177f4324a18871cbb011)

**fixed alerts:**

* 1 for Variable not declared before use

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**